Update getting started page for vSphere

kerneltime opened the pull request:
Deprecate kube-up
Add information for kubernetes-anywhere

jihdeh committed:
Update nodeJS code to supported es6 version LTS

Nodejs now support some es6 features, on it's latest stable version 4.5.
https://nodejs.org/dist/latest-v4.x/docs/api/synopsis.html

Commit:
Clarifying the purpose and intended use of a ConfigMap

jihdeh committed:
Merge branch 'master' into patch-1

Commit:
Merge branch 'master' into patch-1

steveperry-53 committed:
Merge branch 'master' into patch-1

Commit:
Added a Recycler Pod Template Example

Recycle pod template example is missing in the documentation.

That's why it is now added.

newswangerd committed:
made some minor fixes to the centos guide to make it a little easier for beginners to follow

newswangerd committed:
Merge branch 'master' into improve_centos_guide

newswangerd committed:
Merge branch 'master' into improve_centos_guide

berrange committed:
getting-started: replace 'yum' with 'dnf' in Fedora docs

dnf has replaced yum as the standard package install tool
in all versions of Fedora that are currently supported.

Signed-off-by: Daniel P. Berrange <berrange@redhat.com>

berrange committed:
getting-started: don't use updates-testing repo for Fedora

There is no reason to tell people to enable updates-testing
repo for Fedora by default. The latest stable packages are
already in the main repos and updates-testing should only be
used by people wishing to test & report quality of pre-released
software, not general Kubnetes users.

Also remove the link to Koji builds - any Koji build which
is working well enough to user will be submitted as an update.
Telling users to install packages straight from Koji will
lead to them using potentially unstable / buggy builds.

Signed-off-by: Daniel P. Berrange <berrange@redhat.com>

berrange committed:
getting-started: remove note about installing iptables on Fedora

The install of kubernetes pulls in docker, which has a direct
dependancy on iptables. So there is no need to tell people to
yum install iptables manually.

Signed-off-by: Daniel P. Berrange <berrange@redhat.com>

berrange committed:
getting-started: only list config vars that need changing

The KUBE_LOGTOSTDERR, KUBE_LOG_LEVEL and KUBE_ALLOW_PRIV
settings will already have correct values out of the box,
so no need to tell people to set them.

Signed-off-by: Daniel P. Berrange <berrange@redhat.com>

berrange committed:
getting-started: update to assume port numbers for etcd >= 2.0

The docs still refer to etcd using port 4001, but in all
currently released Fedora versions it will be using 2379

Signed-off-by: Daniel P. Berrange <berrange@redhat.com>

berrange committed:
getting-started: remove note to create /var/run/kubernetes

The /var/run/kubernetes directory is already created by
installation of the kubernetes RPM with correct user
and group ownership and permissions.

Signed-off-by: Daniel P. Berrange <berrange@redhat.com>

berrange committed:
getting-started: add instruction to install flannel

The Fedora multi-node doc need to tell users to install
flannel, since previous setup won't have brought it in.

Signed-off-by: Daniel P. Berrange <berrange@redhat.com>

agonzalezro committed:
Use --decode for base64 command

`-d` is used for the Linux version of the command but in the Mac/BSD version they use `-D`.

Using `--decode` we are sure that the flag is compatible with both.

harryge00 committed:
Update dns.md: add details about subdomain.

newswangerd committed:
Merge branch 'master' into improve_centos_guide

newswangerd committed:
Merge branch 'master' into improve_centos_guide

brendandburns committed:
Add directions for using Azure Container Registry.

berrange committed:
getting-started: note that Fedora can use VMs or bare metal

Although the Fedora install docs are under the "Bare Metal"
section in the docs, the instructions provided work fine
for Fedora installs in virtual machines

Signed-off-by: Daniel P. Berrange <berrange@redhat.com>

rivimont committed:
clarification on type LoadBalancer for exposing

Took me a bit of time to learn that type LadBalancer wasn't fully working in an OpenStack environment as I was going through this.

rivimont committed:
Merge branch 'master' into patch-1

rivimont committed:
minor changes

moved text under command and changed wording based on some feedback. With regards to mentioning cloud providers. I think a lot of this documentation in general assumes a public cloud like Google so it may be worthwhile to mention a specific private cloud where this isn't implemented.

rivimont committed:
minor text change

rivimont committed:
Merge branch 'master' into patch-1

foxish committed:
Updated link and fixed typo

jaredbhatti committed:
Merge branch 'master' into patch-1

jaredbhatti committed:
Merge branch 'master' into patch-1

jihdeh committed:
Merge branch 'master' into patch-1

CaoShuFeng committed:
[kubenet] add description about non-masquerade-cidr

For kubenet, users should run kubelet with argument
--non-masquerade-cidr, or pods may have problem when access
"10.0.0.0/8" cidr. Also, if --pod-cidr is set outside the scope
of --non-masquerade-cidr, node will do ip masquerade for ingress
packets, which is not excepted.

rivimont committed:
Merge branch 'master' into patch-1

Hades32 committed:
allow copy&paste of sample code

multi-line code samples should not be prefixed with stuff that makes copy&paste hard!

Commit:
Adding Stackpoint.io as a turn-key provider.

Change-Id: I017e314f67ca3945f658f834ae5b908f30ddd65a

xsgordon committed:
Provide valid --cloud-providers

Provide valid --cloud-providers, vagrant and openshift were listed but is not actually a valid cloud provider in this context in the current code base while a number of other valid options were omitted including azure, cloudstack, openstack, photon and vsphere.

CaoShuFeng committed:
Merge branch 'master' into non-masquerade-cidr

CaoShuFeng committed:
[authorization] update doc about roleRef

This update is made according to this commit in code:
https://github.com/kubernetes/kubernetes/commit/8c788233e778f3b0ebef560762c1433c12ea1d43

It has already be released  with k8s v1.5, so it's reasonable to
update the doc now.

derekwaynecarr committed:
quota by storage class

SEJeff committed:
Linkify issue for audit support in kubernetes

CaoShuFeng committed:
Update authorization.md

fix role name

CaoShuFeng committed:
Merge branch 'master' into roleRef

bibryam committed:
Fixed wrong URL

bibryam committed:
Fixed typos

rivimont committed:
Merge branch 'master' into patch-1

Commit:
Update header

Updates header to remove hamburger on desktop, set 100px margin, and
adds 100px margin to body.

CaoShuFeng committed:
Merge branch 'master' into roleRef

craigbox committed:
Change vertical alignment of code blocks

Things looked odd due to the padding on code blocks; they sat 2px above the text surrounding them.  This PR will fix that by aligning inline code blocks, and the text surrounding them, on their baseline.

mbohlool committed:
Mention OpenAPI in API docs

Commit:
Merge remote-tracking branch 'refs/remotes/kubernetes/master' into UC-design-patch

Commit:
Updates logo

janetkuo committed:
Add left nav for apps API group

jaredbhatti committed:
Merge branch 'master' into apps-nav

jaredbhatti committed:
Merge branch 'master' into dns-doc

jaredbhatti committed:
Merge branch 'master' into linkify-github-issue

jaredbhatti committed:
Merge branch 'master' into quota-by-storage-class

Commit:
Merge branch 'master' into UC-design-patch

Commit:
Merge remote-tracking branch 'refs/remotes/kubernetes/master' into UC-design-patch

Commit:
Merge remote-tracking branch 'refs/remotes/kubernetes/master' into UC-design-patch

Commit:
Minor header change

Change "Try Kubernetes" link point to /docs/tutorials/kubernetes-basics/
instead of "Hello Node"
Reduce font weight in links across the top.

Commit:
Merge branch 'UC-design-patch' of https://github.com/Nanopy/kubernetes.github.io into UC-design-patch

jaredbhatti committed:
Merge branch 'master' into UC-design-patch

andrewwatson committed:
Header displayed twice

It was displaying the header twice

Commit:
Merge branch 'master' into UC-design-patch

jimmycuadra committed:
Fix the formatting of bullet lists on the kubelet auth page.

Commit:
Merge branch 'master' into UC-design-patch

jbeda committed:
Merge branch 'master' into patch-1

Hades32 committed:
Merge branch 'master' into patch-1

benbalter committed:
add explicit titles to docs

gunjan5 committed:
Update Calico docs link

gunjan5 committed:
Merge branch 'master' into calico-addon-link

benbalter committed:
add notitle attribute to pages that shouldnt have a title

Commit:
make the docs content escaped from liquid template brace tag

shettyg committed:
networking.md: Add OVN as a networking plugin option.

OVN is an opensource network virtualization solution
developed by the Open vSwitch community.  It lets one
create logical switches, logical routers, stateful
ACLs, load-balancers etc to build different virtual
networking topologies.  The project has a specific
Kubernetes plugin and documentation at
https://github.com/openvswitch/ovn-kubernetes.

jbeda committed:
Merge branch 'master' into patch-1

joshrosso committed:
statefulsets.md: Fix typos and make some statements more concise

brendandburns committed:
Merge branch 'master' into acr

brendandburns committed:
Merge branch 'master' into acr

joshrosso committed:
services: Clearer definition of types; typo fixes

This commit attempts to make more clear the differences between service
types by ensuring definitions are consistent and concise. It also aims
to fix a few typos and grammatical issues.

brendandburns committed:
Merge pull request #1734 from brendandburns/acr

Add directions for using Azure Container Registry.

Commit:
Merge branch 'master' into escape-from-liquid-template

tianshapjq committed:
delete the question mark

1. add "been" right before restarted.
2. delete the question mark at the end of the sentence.

benbalter committed:
use in-page titles to generate sidebar nav

jaisonpjohn committed:
More Description: use the minikube's built-in Docker daemon

janetkuo committed:
Merge pull request #1989 from tianshapjq/patch-1

delete the question mark

benbalter committed:
Merge branch 'master' into inline-titles

bibryam committed:
Merge branch 'master' into wrong-url

Commit:
getting-started-guides: remove conformance column

This column is sort of pointless as the only data points are two year
old gists.

cychiang committed:
Inconsistent commands in section (5/5)

The original one is: "kops update cluster useast1.dev.awsdata.com --yes"
However, the previous example used useast1.dev.example.com to create configuration file.
So, the "awsdata" should be changed into example.

Commit:
delete italic style

gunjan5 committed:
Merge branch 'master' into calico-addon-link

chentao11596 committed:
change to new config

bibryam committed:
Small typos fixed

xilabao committed:
add reference to rbac & psp

nlamirault committed:
Fix kube-dns component name

benbalter committed:
tell netify to disable the plugin whitelist for now

benbalter committed:
bump to pages gem v112

benbalter committed:
semantically version jekyll-include-cache

jessfraz committed:
Merge pull request #1994 from benbalter/inline-titles

Use in-page titles to generate sidebar nav

devin-donnelly committed:
Merge branch 'master' into calico-addon-link

devin-donnelly committed:
Merge pull request #1971 from gunjan5/calico-addon-link

Update Calico docs link

devin-donnelly committed:
Merge pull request #2004 from nlamirault/patch-1

Fix kube-dns component name

BaluDontu commented on docs/getting-started-guides/vsphere.md:
````diff
@@ -5,27 +5,77 @@ assignees:
+```
+
+* Provide the cloud config on each node to kubelet, apiserver and controller manager via ```--cloud-config=<path to file>```. Cloud config [template can be found at kubernetes-anywere] (https://github.com/kubernetes/kubernetes-anywhere/blob/master/phase1/vsphere/vsphere.conf)
+* Set the cloud provider via ```--cloud-config=vsphere``` flag for kubelet, apiserver and controller manager. 
````
I don't see any difference between above 2 points. Can we just have point 1.

kerneltime replied:
fixed

BaluDontu commented on docs/getting-started-guides/vsphere.md:
```diff
@@ -5,27 +5,77 @@ assignees:
 
 ---
 
-The example below creates a Kubernetes cluster with 4 worker node Virtual
```
Under kube-up section, can we add these statements.

kerneltime replied:
done

BaluDontu commented on docs/getting-started-guides/vsphere.md:
```diff
@@ -5,27 +5,77 @@ assignees:
-The example below creates a Kubernetes cluster with 4 worker node Virtual
-Machines and a master Virtual Machine (i.e. 5 VMs in your cluster). This
-cluster is set up and controlled from your workstation (or wherever you find
-convenient).
```
Under kube-up section, can we add these statements.

kerneltime replied:
done

BaluDontu commented on docs/getting-started-guides/vsphere.md:
````diff
@@ -5,27 +5,77 @@ assignees:
+govc vm.change -e="disk.enableUUID=1" -vm=<VMNAME>
+```
+
+* Provide the cloud config on each node to kubelet, apiserver and controller manager via ```--cloud-config=<path to file>```. Cloud config [template can be found at kubernetes-anywere] (https://github.com/kubernetes/kubernetes-anywhere/blob/master/phase1/vsphere/vsphere.conf)
````
kube-anywhere is mispelled.

kerneltime replied:
fixed

BaluDontu commented on docs/getting-started-guides/vsphere.md:
```diff
@@ -5,27 +5,77 @@ assignees:
+
+1. Volumes 
+2. Persistent Volumes
+3. Storage Classes and provisioning of volumes.
```
Shouldn't we have dynamic provisioning instead of "Storage Classes and provisioning of volumes" as these concepts will be closer as kubernetes mentioned.

kerneltime replied:
The provisioning comes under Storage Classes inside k8s documentation.

BaluDontu commented on docs/getting-started-guides/vsphere.md:
```diff
@@ -5,27 +5,77 @@ assignees:
+
+#### Known issues
+
+* [Volumes are not removed from a VM configuration if the VM is down](https://github.com/kubernetes/kubernetes/issues/33061). The workaround is to manually remove the disk from VM config before powering it up.
```
config --> settings? 

As we can remove disks from VM settings on VC.

BaluDontu reviewed (commented):
Overall looks good. But some very minor comments.

steveperry-53 committed:
Create prerequisites appropriate for load balancer.

king-julien committed:
Merge branch 'master' into master

devin-donnelly committed:
Merge pull request #2003 from xilabao/add-reference-to-rbac-psp

add reference to rbac  psp

devin-donnelly committed:
Merge pull request #2000 from xialonglee/cancel-italic-title

Delete italic style markdown symbol

devin-donnelly committed:
Merge pull request #2001 from chentao1596/kubelet-static-pod-config

static-pod: --config should be changed to -pod-manifest-path

devin-donnelly committed:
Merge pull request #1999 from cychiang/patch-1

Inconsistent command in section (5/5)

devin-donnelly committed:
Merge pull request #1997 from philips/remove-conformance-field

getting-started-guides: remove conformance column

devin-donnelly committed:
Merge pull request #1980 from joshrosso/statefulsets-typo-concise

statefulsets.md: Fix typos and make some statements more concise

devin-donnelly committed:
Merge pull request #1983 from joshrosso/services-cleanup-2

services: Clearer definition of types; typo fixes

devin-donnelly committed:
Merge pull request #1974 from xialonglee/escape-from-liquid-template

Make the docs content escaped from liquid template brace tag

devin-donnelly committed:
Merge pull request #1956 from andrewwatson/patch-3

Header displayed twice

xiangpengzhao committed:
Fix #2005: Issue with /docs/user-guide/horizontal-pod-autoscaling/

bibryam committed:
Fixed cricial to critical

luojie233 committed:
Duplicated 'the'

Signed-off-by: Jie Luo <luo612@zju.edu.cn>

luojie233 committed:
fix some typos

Signed-off-by: Jie Luo <luo612@zju.edu.cn>

king-julien committed:
Merge branch 'master' into master

Commit:
Making the Docker version requirement clear and added command to create a new VMSwtich for kube-proxy to use

Commit:
Merge branch 'master' into master

Commit:
Remove accidentally nested {% raw %} tags.

These tags cannot be nested, causing a Liquid syntax error.
The nesting was introduced accidentally by concurrent PRs.

devin-donnelly committed:
Merge pull request #2017 from enisoc/fix-liquid-raw

Remove accidentally nested {% raw %} tags.

jaredbhatti committed:
Merge branch 'master' into branch3

jaredbhatti committed:
Merge pull request #2011 from luojie233/branch3

fix some typos

king-julien committed:
Merge branch 'master' into master

devin-donnelly committed:
Merge branch 'master' into branch2

devin-donnelly committed:
Merge branch 'master' into win_docker_clarif

devin-donnelly committed:
Merge branch 'master' into fix-cm-link

devin-donnelly committed:
Merge pull request #2002 from bibryam/various-typos

Small typos fixed

devin-donnelly committed:
Merge pull request #2006 from steveperry-53/expose

Create prerequisites appropriate for load balancer.

devin-donnelly committed:
Merge pull request #2016 from apprenda/win_docker_clarif

Docker version requirements and adding New VMSwitch

devin-donnelly committed:
Merge pull request #2010 from luojie233/branch2

Duplicated 'the'

devin-donnelly committed:
Merge branch 'master' into fix-cm-link

devin-donnelly committed:
Merge pull request #2008 from xiangpengzhao/fix-cm-link

Fix broken link in the horizontal pod autoscaling doc

devin-donnelly committed:
Merge pull request #1978 from shettyg/docs

networking.md: Add OVN as a networking plugin option.

devin-donnelly committed:
Merge pull request #1961 from InQuicker/kubelet-auth-formatting

Fix the formatting of bullet lists on the kubelet auth page.

devin-donnelly committed:
Merge branch 'master' into UC-design-patch

devin-donnelly committed:
Merge pull request #1951 from Nanopy/UC-design-patch

UC design patch

devin-donnelly committed:
Merge pull request #1947 from janetkuo/apps-nav

Add left nav for apps API group

devin-donnelly committed:
Merge pull request #1933 from kubernetes/mbohlool-patch

Mention OpenAPI in API docs

devin-donnelly committed:
Merge branch 'master' into patch-5

devin-donnelly committed:
Merge pull request #1918 from craigbox/patch-5

Change vertical alignment of code blocks

devin-donnelly committed:
Merge pull request #1907 from bibryam/dns-doc

Fixed typos

devin-donnelly committed:
Merge pull request #1901 from bibryam/wrong-url

Fixed wrong URL

devin-donnelly committed:
Merge branch 'master' into linkify-github-issue

devin-donnelly committed:
Merge pull request #1895 from xsgordon/cloud-provider-update

Provide valid --cloud-providers

devin-donnelly committed:
Merge pull request #1900 from SEJeff/linkify-github-issue

Linkify issue for audit support in kubernetes

devin-donnelly committed:
Merge pull request #1893 from derekwaynecarr/quota-by-storage-class

Document quota by storage class

devin-donnelly committed:
Merge pull request #1888 from CaoShuFeng/roleRef

[authorization] update doc about roleRef

devin-donnelly committed:
Merge branch 'master' into master

devin-donnelly committed:
Merge pull request #1873 from StackPointCloud/master

Adding Stackpoint.io as a turn-key provider.

devin-donnelly committed:
Merge pull request #1835 from Hades32/patch-1

allow copy&paste of sample code

devin-donnelly committed:
Merge branch 'master' into non-masquerade-cidr

devin-donnelly committed:
Merge pull request #1760 from CaoShuFeng/non-masquerade-cidr

[kubenet] add description about non-masquerade-cidr

devin-donnelly committed:
Update quick-start.md

devin-donnelly committed:
Merge branch 'master' into patch-1

devin-donnelly committed:
Merge pull request #1673 from harryge00/patch-2

Update dns.md

devin-donnelly committed:
Merge pull request #1752 from rivimont/patch-1

clarification on type LoadBalancer for exposing

devin-donnelly committed:
Merge pull request #1672 from agonzalezro/patch-1

Use --decode for base64 command

devin-donnelly committed:
Merge pull request #1665 from berrange/fedora-docs-fixes

Fedora docs fixes

devin-donnelly committed:
Merge pull request #1641 from newswangerd/improve_centos_guide

Minor fixes to the centos guide to make it easier for beginners to follow

devin-donnelly committed:
Merge pull request #1636 from pospispa/86-5-add-checks-and-documentation-about-template-pods-for-recycling

Added a Recycler Pod Template Example

devin-donnelly committed:
Update index.md

devin-donnelly committed:
Merge pull request #1302 from jihdeh/patch-1

Update nodeJS code to supported es6 version LTS

devin-donnelly committed:
Merge pull request #1363 from thesteve0/patch-1

Clarifying the purpose and intended use of a ConfigMap

devin-donnelly committed:
Merge pull request #1995 from king-julien/master

More Description: use the minikube's built-in Docker daemon

Commit:
Deprecate kube-up
Add information for kubernetes-anywhere
